### ISSUE TITLE
Incorrect quoting of time for influx clients with a specific time_precision

### ISSUE THREAD

**Issue by dimiii** [palkan/influxer#35]
### Bug report

__System info:__ [Influxdb v1.4.2, influxdb-ruby v0.3.17, Influxer v1.1.3]

__Steps to reproduce:__

1. Use configuration with a specific time precision
```
Influxer.configure do |config|
   ....
  config.time_precision = 's'
end

```
2.  Execute any query with time filtering, for instance: `where(time: time_vals)`, time_vals is an array of Time objects.

__Expected behavior:__ 
Influxer builds a correct query.

__Actual behavior:__ 
Influxer builds a query with an unquoted time parameter. Moreover, the parameter is formatted improperly (not rfc3339 as influx expects).

__Additional info:__ 
https://github.com/palkan/influxer/commit/b7884c1a6a8af46833268c4ff2672a3e1f7180fd#diff-9eee17a71fdc19385d4e9dec5fd4d31bL273
https://github.com/palkan/influxer/blob/78570b883c883c7c2913ab940c4e065cc5993c35/lib/influxer/metrics/quoting/timestamp.rb#L10

**Comment by palkan** [comment]
Looks like that's an intentional and undocumented behaviour (see [PR](https://github.com/palkan/influxer/pull/16#discussion_r56947237)(

Btw, InfluxDB recognizes different time formats, including `epoch_time` in nanoseconds (see [docs](https://docs.influxdata.com/influxdb/v1.4/query_language/data_exploration/#description-of-syntax-120); that exactly what we use.

And since 1.3 we can specify duration suffix explicitly within a query; I'll release fix ASAP

**Comment by palkan** [comment]
@dimiii I pushed fix to `fix/time-precision` branch; could you, please, try it and confirm, that everything works as expected now? 

Also, see https://github.com/palkan/influxer/wiki/Getting-Started#time-precision

**Comment by dimiii** [comment]
@palkan 

Yup, it works fine and my query fetches required data as expected. But what if instead of scaling to a nanosecond, influxer will do scaling to `Influxer.config.time_precision`? This is more consistent with user's configuration. Plus, everyone loves seconds and milliseconds as timestamps (except odious influxdb developers), and so it would be nice to see specified time precision in queries. 

If `time_precision` is different from default, but `time_duration_suffix_enabled` is false or default, current version produces a query with a timestamp without any precision (time = 1518258600). So, I would like to propose this version of `quote_timestamp`, it's less prone to misconfiguration errors:

```
    def quote_timestamp(val, client)
      if !TIME_FACTORS.keys.include?(client.time_precision) &&
         !val.is_a?(Numeric)
        warn(
          "Influxer doesn't automatically cast Time and String values " \
          "to '#{client.time_precision}' precision. " \
          "Please, convert to numeric value yourself"
        )
        return val
      end
      
      factor = TIME_FACTORS.fetch(client.time_precision)
      timestamp = factorize_timestamp(val, factor)
      "#{timestamp}#{client.time_precision}"
    end 

 or even

    def quote_timestamp(val, client)
      factor = TIME_FACTORS.fetch(client.time_precision, 'ns')
      timestamp = factorize_timestamp(val, factor)
      "#{timestamp}#{client.time_precision}"
    end    
```

To my mind, `time_duration_suffix_enabled` is redundant here.

**Comment by palkan** [comment]
> time_duration_suffix_enabled is redundant here.

The problem is that duration suffixes has been only introduced in 1.3; so, we can just enable it for everyone.

The solution above is, of course, more elegant; but unfortunately, incompatible with older versions (not so old, btw).

**Comment by dimiii** [comment]
@palkan  Ok, then maybe?
```
    def quote_timestamp(val, client)
      if !TIME_FACTORS.keys.include?(client.time_precision) &&
         !val.is_a?(Numeric)
        warn(
          "Influxer doesn't automatically cast Time and String values " \
          "to '#{client.time_precision}' precision. " \
          "Please, convert to numeric value yourself"
        )
        return val
      end

      return quote_timestamp_with_suffix(val, client.time_precision) if Influxer.config.time_duration_suffix_enabled

      factor = TIME_FACTORS.fetch(client.time_precision)
      factorize_timestamp(val, factor)
    end

    def quote_timestamp_with_suffix(val, time_precision)
      factor = TIME_FACTORS.fetch(time_precision)
      timestamp = factorize_timestamp(val, factor)
      "#{timestamp}#{time_precision}"
    end
```
Or maybe something other. My point is: time precision should be used, if client specifies it in config.

**Comment by palkan** [comment]
Released as 1.1.4.

Thanks for help!

**Comment by dimiii** [comment]
Thank you.
